src/main/java/uk/gov/justice/digital/job/DomainRefreshJob.java

```diff
@@ -27,26 +31,29 @@ public class DomainRefreshJob extends Job implements Runnable {
     private final String curatedPath;
     private final String domainTargetPath;
     protected String domainTableName;
-    private final String domainId;
+    private final String domainName;
     private final String domainOperation;
     private final DynamoDBClient dynamoDBClient;
 
+    private final String domainRegistry;
+
+    private final DataStorageService storage;
+
     @Inject
-    public DomainRefreshJob(JobParameters jobParameters, DynamoDBClient dynamoDBClient) {
-        this.curatedPath = jobParameters.getCuratedS3Path()
-                .orElseThrow(() -> new IllegalStateException(
-                        "curated s3 path not set - unable to create CuratedZone instance"
-                ));
+    public DomainRefreshJob(JobParameters jobParameters, DynamoDBClient dynamoDBClient, DataStorageService storage) {
+        this.curatedPath = jobParameters.getCuratedS3Path();
         this.domainTargetPath = jobParameters.getDomainTargetPath();
         this.domainTableName = jobParameters.getDomainTableName();
-        this.domainId = jobParameters.getDomainId();
+        this.domainName = jobParameters.getDomainName();
+        this.domainRegistry = jobParameters.getDomainRegistry();
         this.domainOperation = jobParameters.getDomainOperation();
         this.dynamoDBClient = dynamoDBClient;
+        this.storage = storage;
     }
 
-    public DomainRefreshService refresh() {
+    public DomainService refresh() {
```

Comment:
Do we need to create a new spark session here? Looks like this method just creates an instance of `DomainService`, in which case we can probably remove this method (along with it's weird name) and replace it with a call to the `DomainService` constructor in the `run` method.

Reply:
removed Sparksession and called inside the constructor of DomainExecutor

Reply:
Ok, can you also remove the refresh method as requested since it's not needed. Just call the constructor directly instead.